### PR TITLE
Issue 30: modularise running of `baselinenowcast` pipeline

### LINKED ISSUE #30: Attempt to modularize the `baselinenowcast_pipeline` across `tar_map()`
**Goal**
It would be ideal if we could take bits of the paper analysis pipeline and use the structure for the coding portion of a vignette demonstrating how to vary the default options (e.g. borrow delays, uncertainty estimation, vary training data volume, filter by weekday reference dates and combine etc). Currently, the paper codebase uses wrapper functions `run_baselinenowcast_pipeline()` and `baseline_nowcast_pipeline()` with arguments and if else statements for what to pass to the modular `baselinenowcast` functions. Ideally, we would break this up and expose the intermediate components within each step of the `tar_map()` loop. 

**Solution**
Rewrite the `gen_noro_nowcasts_targets.R` and `gen_covid_nowcast_targets.R` which loop over individual nowcast dates and model specs, and expose each intermediate element as part of the tar map, rather than running in a wrapper function with ifelse statements.

### PR DESCRIPTION
**Pull request by kaitejohnson**
## Description

This PR closes #30. It makes the following changes:
- it deprecates the wrapper functions `run_baselinenowcast_pipeline()` and `baseline_nowcast_pipeline()`
- it puts the logic in these functions directly into the targets pipeline workflow (the advantage of this being we can now examine individual components of the baselinenowcast pipeline). These are now in `gen_noro_nowcasts_targets` and `gen_covid_nowcasts_targets`
- It modifies the `tar_map` for the norovirus nowcasts -- it will now generate a probabilistic nowcast using the base model (all days) and then individually for each weekday, using two different lengths of historical training volume. These are later aggregated so we would have nowcasts for all weekdays, but they are run separately. This seemed like the maximally modular approach in my head (and avoids introducing a second inner loop to map together the weekday nowcasts)
- Note that while I tested the functionality of this for generating the correct mappings, we still need to implement the change in the `baselinenowcast` package to handle asymmetric reporting triangles ( see [here]( https://github.com/epinowcast/baselinenowcast/pull/94)) 

I think in this instance it would be helpful for you to review @seabbs to make sure we are on the same page about modularity

## Checklist

- [X] My PR is based on a package issue and I have explicitly linked it.
- [X] I have included the target issue or issues in the PR title in the for Issue(s) *issue-numbers*: PR title
- [X] I have read the [contribution guidelines](https://github.com/epinowcast/.github/blob/main/CONTRIBUTING.md).
- [X] I have tested my changes locally.
- [X] I have added or updated unit tests where necessary.
- [X] I have updated the documentation if required.
- [X] My code follows the established coding standards.
- [X] I have added a news item linked to this PR.
- [X] I have reviewed CI checks for this PR and addressed them as far as I am able.

### CODE REVIEW

**Comment by kaitejohnson** [comment]
@seabbs Going to merge this though had thought you might have wanted to review. We will have to revise per adjusted function flow.

**Comment by seabbs** [comment]
I thought we weren't doing review flow here? I think my notifications are off or something...

**Comment by kaitejohnson** [comment]
> I thought we weren't doing review flow here? I think my notifications are off or something...
I had thought you specifically wanted to review this one, to make sure we were on the same page about what it meant to modularise the pipeline.

**Comment by kaitejohnson** [comment]
> What I thought we had discuss was coming up with scripts/functions for each option or combination of options (so you have a piped thing that represents one model at the end of the day. I think this is fine for now and we can revisit down the line.

I think it would be piped if we were to write it into a script format, but since we are using targets each output needs to get passed as an input to the next modular function

**Comment by kaitejohnson** [comment]
Or did you envision and I misunderstood, a wrapper function that had internal piping?
